sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala

```diff
@@ -188,17 +194,26 @@ class UnsupportedOperationsSuite extends SparkFunSuite with SQLHelper {
     expectedMsgs = Seq("Complete"))
 
   // FlatMapGroupsWithState(Update) in streaming with aggregation
-  for (outputMode <- Seq(Append, Update, Complete)) {
+  for (outputMode <- Seq(Update, Complete)) {
     assertNotSupportedInStreamingPlan(
       "flatMapGroupsWithState - flatMapGroupsWithState(Update) on streaming relation " +
         s"with aggregation in $outputMode mode",
       TestFlatMapGroupsWithState(
         null, att, att, Seq(att), Seq(att), att, null, Update, isMapGroupsWithState = false, null,
         Aggregate(Seq(attributeWithWatermark), aggExprs("c"), streamRelation)),
       outputMode = outputMode,
-      expectedMsgs = Seq("flatMapGroupsWithState in update mode", "with aggregation"))
+      expectedMsgs = Seq("Multiple stateful operators", "Update", "Complete"))
   }
 
+  assertNotSupportedInStreamingPlan(
+    "flatMapGroupsWithState - flatMapGroupsWithState(Update) on streaming relation " +
+      s"with aggregation in Append mode",
+    TestFlatMapGroupsWithState(
+      null, att, att, Seq(att), Seq(att), att, null, Update, isMapGroupsWithState = false, null,
```

Comment:
I am not sure what is the semantics of setting output mode to Update on flatMapGroupsWithState but it does not match the Append output mode below.

Reply:
Sorry I didn't really get it and I lack the context here. Do you mean flatMapGroupWithState(Update) with aggregation and output mode Append doesn't make sense? I think before there is a test on each output mode: https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala#L191

The reason I separate (Update, Complete) and Append is with the change, (Update, Complete) will be captured before the original check, so their error message is changed. But in Append mode it is still blocked by the original error message.

Reply:
This is one of the reason I want to see the allowed operations be more restrictive rather than saying op A - op B is supported with some modes, op B - op A is supported with some modes, etc. 

The major reason I blocked multiple stateful operators at all was that we figured out stream-stream outer join followed by another stream-stream outer join is not working, and I realized that is not only the case. Similar risks are everywhere and it's really hard to think hard on all matrix.

That said, unfortunately, the case we do not fail in unsupported operation checker does not always mean we guarantee the case to work properly. We probably have checked in some cases, but I don't believe it's backed by theory (otherwise we should have known the limit of global watermark much earlier).